polaris-for-stylelint/src/index.js

```diff
@@ -0,0 +1,38 @@
+const fs = require('fs');
+const path = require('path');
+
+const tokenGroupsDir = path.join(__dirname, './src/tokens/token-groups');
+
+/**
+ * Allowed Polaris token custom properties.
+ *
+ * Result: ['--p-background', '--p-text', etc...]
+ */
+const polarisTokenCustomProperties = Array.from(
+  new Set(
+    fs
+      .readdirSync(tokenGroupsDir)
+      .map((file) => {
+        const tokenGroup = require(path.join(tokenGroupsDir, file));
+
+        return Object.keys(tokenGroup).map((token) => `--p-${token}`);
+      })
+      .flat(),
+  ),
+);
+
+module.exports = {
+  plugins: ['@shopify/custom-properties-allowed-list'],
```

Comment:
Having one plugin per package feels like it's going to get very noisy. Would it be worth consolidating all the plugins into one package? https://github.com/Shopify/web-configs/tree/main/packages/stylelint-plugin is a demo of how you can have many plugins in one package

Reply:
I don't anticipate us creating (m)any more one off Stylelint plugins. `@shopify/custom-properties-allowed-list` is intentionally isolated from `polaris-for-stylelint` as it is consumed in `polaris-react` and `polaris-for-stylelint` with different rulesets (and is Polaris agnostic). Although, I'm quite inspired by the `web-configs/stylelint-plugin` architecture! How would you suggest we structure `polaris-for-stylelint` to contain `custom-properties-allowed-list`? Would it be similar to [web-configs/stylelint-plugin/rules/content-no-strings](https://github.com/Shopify/web-configs/tree/main/packages/stylelint-plugin/rules/content-no-strings)?

Reply:
> Would it be similar to [web-configs/stylelint-plugin/rules/content-no-strings](https://github.com/Shopify/web-configs/tree/main/packages/stylelint-plugin/rules/content-no-strings)?

Yep exactly.

>  [the rule in @shopify/custom-properties-allowed-list] is consumed in polaris-react and polaris-for-stylelint with different rulesets (and is Polaris agnostic).

That doesn't seem like a critical reason for it being a separate package - It seems fine if polaris-react consumed polaris-for-stylelint (containing the custom-properties-allowed-list) but with enabled the rule with a slightly different  configuration to the default that the polaris-for-stylelint extendable config provided.

Reply:
Updated the PR description as well